google/resource_bigtable_gc_policy.go

```diff
@@ -0,0 +1,234 @@
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := d.Get("instance_name").(string)
+	c, err := config.bigtableClientFactory.NewAdminClient(project, instanceName)
+	if err != nil {
+		return fmt.Errorf("Error starting admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	gcPolicy, err := generateBigtableGCPolicy(d)
+	if err != nil {
+		return err
+	}
+
+	tableName := d.Get("table").(string)
+	columnFamily := d.Get("column_family").(string)
+
+	if err := c.SetGCPolicy(ctx, tableName, columnFamily, gcPolicy); err != nil {
+		return err
+	}
+
+	table, err := c.TableInfo(ctx, tableName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving table. Could not find %s in %s. %s", tableName, instanceName, err)
+	}
+
+	for _, i := range table.FamilyInfos {
+		if i.Name == columnFamily {
+			d.SetId(i.GCPolicy)
+		}
+	}
+
+	return resourceBigtableGCPolicyRead(d, meta)
+}
+
+func resourceBigtableGCPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := d.Get("instance_name").(string)
+	c, err := config.bigtableClientFactory.NewAdminClient(project, instanceName)
+	if err != nil {
+		return fmt.Errorf("Error starting admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	name := d.Get("table").(string)
+	ti, err := c.TableInfo(ctx, name)
+	if err != nil {
+		log.Printf("[WARN] Removing %s because it's gone", name)
+		d.SetId("")
+		return fmt.Errorf("Error retrieving table. Could not find %s in %s. %s", name, instanceName, err)
+	}
+
+	for _, fi := range ti.FamilyInfos {
+		if fi.Name == name {
+			d.SetId(fi.GCPolicy)
+			break
+		}
+	}
+
+	d.Set("project", project)
+
+	return nil
+}
+
+func resourceBigtableGCPolicyDestroy(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instanceName := d.Get("instance_name").(string)
+	c, err := config.bigtableClientFactory.NewAdminClient(project, instanceName)
+	if err != nil {
+		return fmt.Errorf("Error starting admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	if err := c.SetGCPolicy(ctx, d.Get("table").(string), d.Get("column_family").(string), bigtable.NoGcPolicy()); err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func generateBigtableGCPolicy(d *schema.ResourceData) (bigtable.GCPolicy, error) {
+	var policies []bigtable.GCPolicy
+	mode := d.Get("mode").(string)
+	ma, aok := d.GetOk("max_age")
+	mv, vok := d.GetOk("max_version")
+
+	if !aok && !vok {
+		return bigtable.NoGcPolicy(), nil
+	}
+
+	if mode == "" && aok && vok {
+		return nil, fmt.Errorf("If multiple policies are set, mode can't be empty")
+	}
+
+	if aok {
+		d, err := strconv.Atoi(ma.(map[string]interface{})["days"].(string))
```

Comment:
I think that `ma` needs to be converted to `[]interface{}`, and the `0` element should be a `map[string]interface{}`.

Reply:
I've change, not sure I should check if slice has at least one element, shouldn't go in this statement if value isn't set in .tf files